test/shanghai-opcodes/ShanghaiOpcodes.js

```diff
@@ -0,0 +1,51 @@
+const { expect } = require("chai");
+const { ethers } = require("hardhat");
+
+describe("Shanghai opcodes tests", function () {
+  let signers;
+  let shanghaiContract;
+
+  before(async function () {
+    signers = await ethers.getSigners();
+
+    const factory = await ethers.getContractFactory('ShanghaiOpcodes');
+    shanghaiContract = await factory.deploy();
+  });
+
+  it("should be able to execute opShl()", async function () {
+    const res = await shanghaiContract.opShl(2, 10);
+    // shift left
+    expect(res).to.equal(40);
+  });
+
+  it("should be able to execute opShr()", async function () {
+    const res = await shanghaiContract.opShr(2, 500);
+    // shift right
+    expect(res).to.equal(125);
+  });
+
+  it("should be able to execute opSar()", async function () {
+    const res = await shanghaiContract.opSar(1, 2);
```

Comment:
This would be a bit more convincing with a different target number than `2`.  Like the examples above.